### PR TITLE
Feature/makefile

### PR DESCRIPTION
**Pull request by vsaveliev**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,79 @@
+.PHONY: all
+all: push
+
+BUILDTAGS=
+
+# Use the 0.0.0 tag for testing, it shouldn't clobber any release builds
+APP=codegen
+PROJECT=github.com/k8s-community/codegen
+RELEASE?=0.0.0
+GOOS?=darwin
```
Comment (takama) [inline Makefile:10]: Maybe linux is more friendly by default?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+.PHONY: clean
+clean:
+	rm -f ./bin/${GOOS}/${APP}
+
+HAS_GLIDE := $(shell command -v glide;)
+HAS_LINT := $(shell command -v golint;)
+
+.PHONY: bootstrap
+bootstrap:
+ifndef HAS_GLIDE
```
Comment (takama) [inline Makefile:74]: We should have agreement in choice of package manager. Is is glide or dep?

**Review by takama — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (rumyantseva) [reply Makefile:74]: Maybe it's possible to add code to the Makefile for both of them and to give the opportunity to choose?

Comment (takama) [reply Makefile:74]: Right, but this is usable for template, not for codegen utility

**Comment by rumyantseva** [comment]
What do you think about usage of gometalinter? 
Previously, we used more linters (for example, we used errcheck etc), gometalinter contains more helpful things - https://github.com/alecthomas/gometalinter 

Another helpful thing: https://staticcheck.io

**Comment by rumyantseva** [comment]
Our makefile becomes complicate. So, I think, we need to write some documentation (probably, in README.md) about Makefile.

**Pull request closed without merging**